src/library/scala/collection/immutable/NVector.scala

```diff
@@ -98,6 +107,14 @@ sealed abstract class NVector[+A](protected[this] final val prefix1: Arr1)
 
   protected[this] def tinyAppendLimit: Int = 4 + vectorSliceCount
 
+  override final def length: Int =
+    if(this.isInstanceOf[BigNVector[_]]) this.asInstanceOf[BigNVector[_]].length0
+    else prefix1.length
+
+  override final def iterator: Iterator[A] =
+    if(this.isInstanceOf[NVector0.type]) NVector.emptyIterator
+    else new NVectorIterator(this, length, vectorSliceCount)
```

Comment:
what if you'd do:
```
length match {
  case 0 => NVector.emptyIterator
  case 1 => Iterator.single(1)
  case n => new NVectorIterator(this, n, vectorSliceCount)
}
```

Reply:
That's exactly the kind of pattern you need to avoid. NVector1 used ArrayIterator before but it's detrimental for performance with mixed-size warmup because you don't get monomorphic dispatch to the iterator methods anymore.

Reply:
@szeiger Makes sense—I should've thought of that myself. :)